### PR TITLE
[Aptos Framework][Account] added revoke signer capability function

### PR DESCRIPTION
**Pull request by chloeqjz**
### Description
added function to revoke signer capability

### Test Plan

unit tests

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4195)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 1]
```diff
@@ -589,6 +606,30 @@ TODO bring back with generic rotation capability
         create_authorized_signer(&charlie, signer::address_of(&alice));
     }
 
+    #[test(bob = @0x345)]
+    public entry fun test_valid_revoke_signer_capability(bob: signer) acquires Account {
+        let pk = x"f66bf0ce5ceb582b93d6780820c2025b9967aedaa259bdbb9f3d0297eced0e18";
+        let alice = create_account_from_ed25519_public_key(pk);
+        create_account(signer::address_of(&bob));
+
+        let valid_signature = x"cd181d65eb31193dcf1627fc0cc04208f66e7f243facc840830eaa458b176de570f73b661c127d98bc276c5a07ab242734b4d656163a86803561c0b9d9d01d0c";
```
Comment (alinush) [inline aptos-move/framework/aptos-framework/sources/account.move:615]: Can you add instructions here on how others can reproduce this signature? What did you use?

**Review by alinush — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/account.move (hunk 1 above)
Comment (chloeqjz) [reply aptos-move/framework/aptos-framework/sources/account.move:615]: I used Rust to mock a transaction - will add a note!

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 2]
```diff
@@ -330,11 +339,15 @@ module aptos_framework::account {
...
         // update the existing signer capability offer or put in a new signer capability offer for the current account
-        if (option::is_some(&account_resource.signer_capability_offer.for)) {
-            option::swap(&mut account_resource.signer_capability_offer.for, recipient_address);
-        } else {
-            option::fill(&mut account_resource.signer_capability_offer.for, recipient_address);
-        };
+        option::swap_or_fill(&mut account_resource.signer_capability_offer.for, recipient_address);
+    }
+
+    public fun revoke_signer_capability(account: &signer, to_be_revoked_address: address) acquires Account {
```
Comment (areshand) [inline aptos-move/framework/aptos-framework/sources/account.move:345]: make this an entry function as well?

Comment (chloeqjz) [reply aptos-move/framework/aptos-framework/sources/account.move]: just updated!

Comment (chloeqjz) [reply aptos-move/framework/aptos-framework/sources/account.move]: good point!

**Review by davidiw — APPROVED** [review]

**Pull request merged**